torch/csrc/jit/script/compiler.cpp

```diff
@@ -1931,9 +1937,24 @@ std::vector<Argument> parseArgsFromDecl(Decl decl, bool is_method) {
   size_t i = is_method ? 1 : 0;
   for (; i < decl.params().size(); ++i) {
     auto decl_arg = decl.params()[i];
-    auto arg = Argument(decl_arg.ident().name(), parseTypeFromExpr(decl_arg.type()),
-                        /*N =*/at::nullopt, /*default_value =*/at::nullopt,
-                        /*kwarg_only =*/false);
+    at::optional<IValue> default_value = at::nullopt;
+    if (decl_arg.has_default()) {
+      // If the Param has a default value, convert it to an IValue
+      Const default_const = decl_arg.default_value();
```

Comment:
I am not sure what is going on here. I would rather not support default arguments from our string frontend yet, as it requires implementing a bunch more infrastructure than we currently have.